docs/contributing.rst

```diff
@@ -0,0 +1,200 @@
+
+1. Fork the ``fmu-dataio`` repo on Github equinor to your personal user
+2. Clone your fork locally:
+
+.. code-block:: bash
+
+    $ git clone git@github.com:your_name_here/fmu-dataio
+    $ cd fmu-dataio
+    $ git remote add upstream git@github.com:equinor/fmu-dataio
+    $ git remote -v
+    origin	git@github.com:your_name_here/fmu-dataio (fetch)
+    origin	git@github.com:your_name_here/fmu-dataio (push)
+    upstream	git@github.com:equinor/fmu-dataio (fetch)
+    upstream	git@github.com:equinor/fmu-dataio (push)
+
+
+3. Install your local copy into a virtualenv. Using python 3, this is how you set
+up your fork for local development (first time):
+
+.. code-block:: bash
+
+    $ cd <fmu-dataio>
+    $ python -m venv .
+    $ source bin/activate
+    $ pip install pip -U
+    $ pip install ".[dev,docs]"
+    $ pytest  # No tests should fail. (exit code 0)
+
+4. Create a branch for local development:
+
+.. code-block:: bash
+
+    $ git checkout -b name-of-your-bugfix-or-feature
+
+Now you can make your changes locally.
+
+5. When you're done making changes, check that your changes pass ruff and the tests:
+
+.. code-block:: bash
+
+    $ ruff check .
+    $ pytest
+
+6. If you want to edit C code, take contact with the author for detailed instructions.
+
+
+7. Commit your changes (see below) and push your branch to GitHub:
+
+.. code-block:: bash
+
+    $ git add .
+    $ git commit -m "AAA: Your detailed description of your changes."
+    $ git push origin name-of-your-bugfix-or-feature
+
+8. Submit a pull request through the Github website.
+
+
+Writing commit messages
+-----------------------
+The following takes effect from year 2021.
+
+Commit messages should be clear and follow a few basic rules. Example:
+
+.. code-block:: text
+
+    ENH: add functionality X to numpy.<submodule>.
+
+The first line of the commit message starts with a capitalized acronym
+(options listed below) indicating what type of commit this is.  Then a blank
+line, then more text if needed.  Lines shouldn't be longer than 72
+characters.  If the commit is related to a ticket, indicate that with
+``"See #3456", "Cf. #3344, "See ticket 3456", "Closes #3456"`` or similar.
+
+Read `Chris Beams hints on commit messages <https://chris.beams.io/posts/git-commit/>`_.
+
+Describing the motivation for a change, the nature of a bug for bug fixes or
+some details on what an enhancement does are also good to include in a commit message.
+Messages should be understandable without looking at the code changes.
+A commit message like FIX: fix another one is an example of what not to do;
+the reader has to go look for context elsewhere.
+
+Standard acronyms to start the commit message with are:
+
+.. code-block:: text
+
+    API: an (incompatible) API change (will be rare)
+    BLD: change related to building fmu-dataio
+    BUG: bug fix
+    CLN: code cleanup, maintenance commit (refactoring, typos, PEP, etc.)
+    DEP: deprecate something, or remove a deprecated object
```

Comment:
Maybe @perolavsvendsen and @jcrivenaes have might feel differently, but I don't think we should bring numpy-style commit prefixing into this repo

Reply:
We have not made this convention for this repo (we have not actually discussed it either, yet).